### PR TITLE
kn export defect to honor mode

### LINKED ISSUE #1087: Export command doesn't honour --mode flag
### Bug report

When `--mode` flag is specified for ksvc without `--with-revisions` flag that output format is plain Knative Service yaml instead of our custom `Export` type.

```
➜  ~ kn service export hello -o yaml --mode export                 
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  creationTimestamp: null
  name: hello
spec:
  template:
    metadata:
      annotations:
        client.knative.dev/user-image: docker.io/dsimansk/helloworld
      creationTimestamp: null
      name: hello-djrrc-1
    spec:
      containerConcurrency: 0
      containers:
      - env:
        - name: TARGET
          value: v1
        image: docker.io/dsimansk/helloworld
        name: user-container
        readinessProbe:
          successThreshold: 1
          tcpSocket:
            port: 0
        resources: {}
      enableServiceLinks: false
      timeoutSeconds: 300
status: {}

```
### Expected behavior

`kn service export hello -o yaml --mode export` - output is printed in requested `Export` format.

### Steps to reproduce the problem

Exec and check the format.
```
 kn service export hello -o yaml --mode export                 
```

### kn version

HEAD

### Knative (serving/eventing) version

> Nightly

/kind good-first-issue

### PR DESCRIPTION
**Pull request by itsmurugappan**
## Description

Mode is ignored, when --with-revisions flag is not mentioned, this pr makes sure that mode is always honored.

## Changes

* "replay" is the default mode
* replay without revisions will print ksvc
* replay with revisions will print ksvc list
* export mode with/without revisions will return knexport type

## Reference

Fixes #1087 

cc @dsimansk

### CODE REVIEW

**Comment by dsimansk** [comment]
> * "replay" is the default mode

I'm leaning towards `export` mode as default a bit more, since that should be added value of `kn service export` command. I let the other chime in, not really insisting on it. :)

I'm planing to take a closer look at the code later today to provide feedback.

File: pkg/kn/commands/service/export_test.go [hunk 1]
```diff
@@ -56,12 +56,6 @@ func TestServiceExportError(t *testing.T) {
...
 	assert.Error(t, err, "'kn service export' requires output format")
-
-	_, err = executeServiceExportCommand(t, tc, "export", tc.latestSvc.ObjectMeta.Name, "--with-revisions", "-o", "json")
-	assert.Error(t, err, "'kn service export --with-revisions' requires a mode, please specify one of replay|export")
-
-	_, err = executeServiceExportCommand(t, tc, "export", tc.latestSvc.ObjectMeta.Name, "--with-revisions", "--mode", "k8s", "-o", "yaml")
-	assert.Error(t, err, "'kn service export --with-revisions' requires a mode, please specify one of replay|export")
 }
 
 func TestServiceExport(t *testing.T) {
```
Comment (dsimansk) [inline pkg/kn/commands/service/export_test.go:61]: I think we should have a similar set of tests for `service export --mode export` without `--with-revisions` that'd also verity that the fix is working as intended and reflecting `--mode` flag.

File: pkg/kn/commands/service/export_test.go [hunk 2]
```diff
@@ -56,12 +56,6 @@ func TestServiceExportError(t *testing.T) {
 
 	_, err := executeServiceExportCommand(t, tc, "export", tc.latestSvc.ObjectMeta.Name)
 	assert.Error(t, err, "'kn service export' requires output format")
```
Comment (maximilien) [inline pkg/kn/commands/service/export_test.go:58]: Can we have a simple test verifying no latest revision in export?

**Review by maximilien — APPROVED** [review]
/ok-to-test

**Comment by dsimansk** [comment]
Hi @itsmurugappan just a quick statu check, any update wrt review comments above?

**Comment by itsmurugappan** [comment]
hi @dsimansk , will make the changes this week. Sorry got side tracked.

File: pkg/kn/commands/service/export_test.go (hunk 1 above)
Comment (itsmurugappan) [reply pkg/kn/commands/service/export_test.go:61]: added this test

**Comment by rhuss** [comment]
Thanks, lets wait until friday for the poll in #1087 for what to use as the default mode, merge this PR and I think since this mode is marked as experimental still, we can create a 0.21.1 afterwards so that we don't have to wait for 0.22 to land.

**Review by dsimansk — APPROVED** [review]
/lgtm

**Comment by dsimansk** [comment]
Ups, I haven't noticed it's already approved and going to be merged on `lgtm`.

@itsmurugappan can you do a follow-up with changelog update please?

**Pull request merged**